### ISSUE TITLE
databaseName property

### ISSUE THREAD

**Issue by carpiediem** [CraZySacX/node-jdbc#53]
I'm not quite sure how to specify my database name and I was hoping you might take a quick look at my code.  When I run reserve(), I don't seem to be getting anything helpful in the callback function.  FYI, this is all in a controller of a sails.js project, since I initially tried to use sails to set up a simple API using an existing JDBC connection.

    var JDBC = require('jdbc');
    var jinst = require('jdbc/lib/jinst');

    if (!jinst.isJvmCreated()) {
      jinst.addOption("-Xrs");
      jinst.setupClasspath(['./drivers/hsqldb.jar',
                            './drivers/derby.jar',
                            './drivers/derbyclient.jar',
                            './drivers/derbytools.jar']);
    }

    var config = {
      url: 'jdbc:sqlserver://mysubdomain.mydomain.net:1433',
      user : 'myusername',
      password: 'mypassword',
      properties: {databaseName: 'frontofficedev'},
      minpoolsize: 10,
      maxpoolsize: 100
    };

    var testpool = null;
    var testconn = null;
     
    var hsqldb = new JDBC(config);

    hsqldb.initialize(function(err) {
      if (err) {
        console.log(err);
      }
    });
     
    var asyncjs = require('async');
     
    module.exports = {
      hi: function (req, res) {
        return res.send("Hi there!");
      },
      
      jdbc: function (req, res) {
        
        console.log("hsqldb = ", hsqldb);
        hsqldb.reserve(function(err, connObj) {
          
          console.log("connObj = ", connObj);
          console.log("err = ", err);
    ...

The CLI displays these log results when I point my browser to http://localhost:1337/advisor/jdbc

    hsqldb =  { _url: 'jdbc:sqlserver://mysubdomain.mydomain.net:1433',
      _props: {},
      _minpoolsize: 10,
      _maxpoolsize: 100,
      _pool: [ undefined, undefined ],
      _reserved: [] }
    connObj =  undefined
    err =  null

The log makes me wonder if I'm not specifying the database correctly.  Or maybe there's a driver issue?  I have successfully connected to and queried the database using SQL Workbench; here's the configuration:

![sqlserver connection profile](https://cloud.githubusercontent.com/assets/5041066/9571372/23124908-4fcf-11e5-9a84-9a7b745d7ace.png)

Thanks for any advice you can give me.  The examples you've given in readme.md are great, but I can't help but think I'm missing something obvious.

**Comment by CraZySacX** [comment]
If you copied and pasted the code you are trying, then the drivers aren't setup correctly.

You should include the sqlserver drives in the jinst section:

```
if (!jinst.isJvmCreated()) {
  jinst.addOption("-Xrs");
  jinst.setupClasspath(['./pathto/sqlserverdriver.jar']);
}
```

**Comment by carpiediem** [comment]
Good catch.  I copied the JAR file that is successfully used by SQL Workbench into the assets/drivers directory of my sails.js project and modified the `jinst` code.

    if (!jinst.isJvmCreated()) {
      jinst.addOption("-Xrs");
      jinst.setupClasspath(['/drivers/sqlserver-jdbc-2.0.jar']);
    }

Unfortunately, I'm getting the same results as before.  I also tried copying the other JAR files into the same directory and using this code.

    if (!jinst.isJvmCreated()) {
      jinst.addOption("-Xrs");
      jinst.setupClasspath(['/drivers/hsqldb.jar',
                            '/drivers/derby.jar',
                            '/drivers/derbyclient.jar',
                            '/drivers/derbytools.jar',
                            '/drivers/sqlserver-jdbc-2.0.jar']);
    }


But still no luck.  Am I missing something?

**Comment by carpiediem** [comment]
I'm still not sure where I went wrong, but I wound up going in a different direction.  [I was able to get sails.js to connect to the database](http://stackoverflow.com/questions/32264207/how-do-i-get-sails-js-to-query-an-existing-database) and do everything I need it to do.

Since this issue is probably based on a bug in my code (not the node-jdbc project), feel free to close it.  Thanks for your quick response to help me out.
